eos-updater-flatpak-installer/main.c

```diff
@@ -544,6 +594,91 @@ parse_installer_mode (const gchar              *mode,
   return TRUE;
 }
 
+static const gchar *
+format_remote_ref_action_type (EosUpdaterUtilFlatpakRemoteRefActionType    action_type,
+                               GError                                    **error)
+{
+  GEnumClass *enum_class = g_type_class_ref (EOS_TYPE_UPDATER_UTIL_FLATPAK_REMOTE_REF_ACTION_TYPE);
+  GEnumValue *enum_value = g_enum_get_value (enum_class, action_type);
+
+  g_type_class_unref (enum_class);
+
+  if (!enum_value)
+    {
+      g_set_error (error,
+                   G_IO_ERROR,
+                   G_IO_ERROR_FAILED,
+                   "Invalid action type %d",
+                   action_type);
+      return NULL;
+    }
+
+  return enum_value->value_nick;
+}
+
+static gboolean
+log_all_flatpak_ref_actions (const gchar  *title,
+                             GHashTable   *flatpak_ref_actions_for_this_boot,
+                             GError      **error)
+{
+  gpointer key, value;
+  GHashTableIter iter;
+
+  g_message ("%s:", title);
+
+  g_hash_table_iter_init (&iter, flatpak_ref_actions_for_this_boot);
+
+  while (g_hash_table_iter_next (&iter, &key, &value))
+    {
+      const gchar *source = (const gchar *) key;
+      GPtrArray *actions = (GPtrArray *) value;
+
+      gsize i;
+
+      g_message("  %s:", source);
+
+      for (i = 0; i < actions->len; ++i)
+        {
+          FlatpakRemoteRefAction *action = g_ptr_array_index (actions, i);
+          const gchar *formatted_action_type = NULL;
+          g_autofree gchar *formatted_ref = NULL;
+
+          formatted_action_type = format_remote_ref_action_type (action->type, error);
```

Comment:
Would we ever expect this to fail? Shouldn’t the `action->type` have been validated earlier?

Reply:
Yeah, it's an enum value... I'd be happy to drop the error handling from that function.